openedx/core/lib/xblock_builtin/xblock_discussion/tests.py

```diff
@@ -0,0 +1,153 @@
+            discussion_category="{category}"
+            discussion_target="{target}"
+        />
+        """.format(category=category, target=target)
+        node = etree.fromstring(xblock_xml)
+
+        patched_load_definition_xml.side_effect = Exception("Irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("Test category", "Test target"),
+        ("Some other category", "Some other target")
+    )
+    def test_legacy_export_format(self, category, target, patched_load_definition_xml):
+        """
+        Test that legacy export XML format can be parsed preserving field values
+        """
+        xblock_xml = """<discussion url_name="82bb87a2d22240b1adac2dfcc1e7e5e4"/>"""
+        xblock_definition_xml = """
+            <discussion discussion_category="{category}" discussion_target="{target}"/>
+        """.format(category=category, target=target)
+        node = etree.fromstring(xblock_xml)
+        definition_node = etree.fromstring(xblock_definition_xml)
+
+        patched_load_definition_xml.return_value = (definition_node, "irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    @mock.patch(DISCUSSION_XBLOCK_LOCATION + ".load_definition_xml")
+    @ddt.unpack
+    @ddt.data(
+        ("Test category", "Test target"),
+        ("Some other category", "Some other target")
+    )
+    def test_legacy_export_format_with_translations(self, category, target, patched_load_definition_xml):
+        """
+        Test that legacy export XML format can be parsed preserving field values
+        """
+        xblock_xml = """<discussion url_name="82bb87a2d22240b1adac2dfcc1e7e5e4"/>"""
+        xmodule_definition_xml = """
+            <discussion discussion_category="{category}" for="{target}"/>
+        """.format(category=category, target=target)
+        node = etree.fromstring(xblock_xml)
+        definition_node = etree.fromstring(xmodule_definition_xml)
+
+        patched_load_definition_xml.return_value = (definition_node, "irrelevant")
+
+        block = DiscussionXBlock.parse_xml(node, self.runtime_mock, self.keys, self.id_gen_mock)
+        self.assertEqual(block.discussion_category, category)
+        self.assertEqual(block.discussion_target, target)
+
+    def test_export_default_discussion_id(self):
+        """
+        Test that default discussion_id values are not exported.
+
+        Historically, the OLX format does not include discussion ID values; instead, the IDs are generated
```

Comment:
I was wrong here, so it's not correct to say "the OLX format does not include discussion ID values". Per the docs, the OLX format can contain either an `id` or a `discussion_id` attribute.

However, those values use `default=UNIQUE_ID` and Studio only exports them if they were explicitly set (either via hand-edited OLX that was imported, or by an author clicking the "duplicate" button in Studio which I believe may explicitly set the discussion_id, which is why the docs say not to do it.).